### PR TITLE
Kimkulling/cleanup after review

### PR DESCRIPTION
**Pull request by kimkulling**
- Add doc + unittest for small class Maybe.

### CODE REVIEW

File: code/Common/Maybe.h [hunk 1]
```diff
@@ -1,29 +1,89 @@
...
         return _val;
     }
 
-private:
     Maybe &operator&() = delete;
+    Maybe(const Maybe &) = delete;
+
+private:
+    T _val;
+    bool _valid;
```
Comment (Skylion007) [inline code/Common/Maybe.h:86]:
```suggestion
    bool _valid = false
```
Then you can simplify the default ctor to 
```
Maybe() = default;
``` 
which is might be a bit more readable.

Comment (kimkulling) [reply code/Common/Maybe.h:86]: Done!

**Review by Skylion007 — APPROVED** [review]

File: code/Common/Maybe.h [hunk 2]
```diff
@@ -82,7 +82,7 @@ struct Maybe {
     Maybe(const Maybe &) = delete;
 
 private:
-    T _val;
+    T _val = false;
```
Comment (Skylion007) [inline code/Common/Maybe.h]: @kimkulling Wrong line. the thing below it should be default initialized.

Comment (kimkulling) [reply code/Common/Maybe.h]: Saw it, already fixed. Thanks for the hint!

**Pull request merged**